lib/wal/wal_omf.h

```diff
@@ -130,6 +161,27 @@ OMF_SETGET(struct wal_filehdr_omf, fh_endoff, 64);
 OMF_SETGET(struct wal_filehdr_omf, fh_cksum, 64);
 
 
+struct wal_rechdr_omf_v1 {
+    uint64_t rh_off;
+    uint32_t rh_flags;
+    uint32_t rh_cksum;
+    uint64_t rh_rid;
+    uint64_t rh_gen;
+    uint32_t rh_type;
+    uint32_t rh_len;
+    uint64_t rh_rsvd;
+} __attribute__((packed,aligned(sizeof(uint64_t))));
```

Comment:
Use attributes defined in compiler.h

Reply:
There is no such thing.  Probably we need an "HSE_OMF_ATTRS" or something that we'd put on all omf structures..

Reply:
```c
#if defined(SUPPORTS_ATTR_PACKED) && defined(SUPPORTS_ATTR_ALIGNED)
#define HSE_OMF_ATTRS __attribute__((packed,aligned(sizeof(uint64_t))))
#else
#error "HSE requires the ability to pack and align on-media structures"
#endif
```

Something like this maybe? Not sure if the error is necessary, but sounds like without this there could be problems.

Reply:
Right.  There are some attributes that if they don't exist then hse likely either wont work or will run poorly.  Would rather know that they don't exist than to silently compile as if everything is ok.  Do we know of any compilers that don't support all the attributes we use?  xlc? icc?

Reply:
No clue...

Need to get HSE compiling on clang first though. Haven't tested it. Would also be nice to move off of gnu11 to c11 although I anticipate a lot of struggle there.

Reply:
Thanks, we can do this cleanup in a different ticket..